src/zope/session/session.py

```diff
@@ -393,35 +403,45 @@ def _sdc(self, pkg_id):
             return zope.component.getUtility(ISessionDataContainer)
 
     def get(self, pkg_id, default=None):
+        """
+        Get session data.
 
-        """See zope.session.interfaces.ISession
+            >>> from zope.publisher.interfaces import IRequest
+            >>> from zope.publisher.http import HTTPRequest
+            >>> from io import BytesIO
+            >>> from zope.session.interfaces import IClientIdManager, IClientId
+            >>> from zope.session.interfaces import ISessionDataContainer
+            >>> from zope.session.http import CookieClientIdManager
+            >>> zope.component.provideUtility(CookieClientIdManager(), IClientIdManager)
+            >>> zope.component.provideAdapter(ClientId, (IRequest,), IClientId)
+            >>> sdc = PersistentSessionDataContainer()
+            >>> zope.component.provideUtility(sdc, ISessionDataContainer, '')
```

Comment:
Maybe hide this within a `.. test-setup`?

Reply:
I don't think I can do that. The .py files are still being run as regular doctests for speed and convenience. `.. testsetup::` expects to find just a block of Python code to execute, but doctest needs the leading `>>>`; the two are in conflict.

Unless there's another way around that?

Reply:
Ahh, I think this will do it:
```rst
.. doctest::
    :hide:

   >>> "code goes here"
```

Reply:
Ah, testsetup is a real directive!  I always thought it was just a ReST comment block of some kind.